src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperationStore.cs

```diff
@@ -49,6 +51,23 @@ public bool IsStoppable
             _serviceProvider = serviceProvider;
         }
 
+        public async Task<ExecutionResult<EgressResult>> ExecuteOperation(IEgressOperation egressOperation)
+        {
+            // Collection Rules do not follow request limits.
+
+            Guid operationGuid = await AddOperation(egressOperation, RequestLimitTracker.Unlimited);
+            EgressEntry entry = null;
+
+            lock (_requests)
```

Comment:
Could you refactor the existing `AddOperation` method to be `Task<EgressEntry> AddOperationInternal(IEgressOperation, string)` and then add back `Task<Guid> AddOperation(IEgressOperation, string)` and have the `ExecuteOperation` method call `AddOperationInternal` to avoid the second lock and dictionary lookup?